packages/fxa-auth-server/lib/routes/oauth/authorization.js

```diff
@@ -129,16 +129,26 @@ module.exports = ({ log, oauthDB, config }) => {
     // Clients must use a single specific redirect_uri,
     // but they're allowed to not provide one and have us fill it in automatically.
     payload.redirect_uri = payload.redirect_uri || client.redirectUri;
-    if (payload.redirect_uri !== client.redirectUri) {
-      log.debug('redirect.mismatch', {
-        param: payload.redirect_uri,
-        registered: client.redirectUri,
-      });
+
+    // Starting in train-248, FxA supports a comma separated list of redirect urls.
+    // However, the client redirectUri must be an exact match, pattern matching
+    // is not supported.
```

Comment:
What's the difference between these redirect URIs and the client redirect URI?

Reply:
The redirect URI here is the one passed from the query params and specified by the relying party oauth authorization flow. It will be checked against the URIs stored in the database for the oauth client.

Reply:
Okay in talking with Barry, the `client.redirectUri` is the value in the DB (could now be a list of values), while the `payload.redirectUri` is what's specified in the request, if any.